### PR TITLE
CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
This is so far just a draft but all feedback (from anyone) is welcome and appreciated!

Currently there is just a outline of the headlines but I will add more content section by section over the coming days.

TODO:
- [x] Non-code contributions
- [x] Config files
- [x] Parser requirements
- [x] Formatting the parsers
- [x] Frontend structure
- [x] State management
- [x] Formatting the frontend

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
@madsnedergaard @tonypls could you guys take a look at this and provide some general feedback? Julien would like to have this done when we release the new app so if I get some early feedback I'll have time to make adjustments.

Right now the fronted structure is missing and I'm not sure if we want to include that if it becomes too complex, it might be a better fit for a more extensive Wiki page.

File: CONTRIBUTING.md
Comment (tonypls) [inline CONTRIBUTING.md:36]: Maybe an example blob would be nice

File: CONTRIBUTING.md
Comment (VIKTORVAV99) [reply CONTRIBUTING.md:63]: Ah yeah that's smart, then we could further link to wiki resources if needed.

Comment (VIKTORVAV99) [reply CONTRIBUTING.md:86]: I added a bit on the structure but I think I want to elaborate on it a little more in the future.

Comment (tonypls) [reply CONTRIBUTING.md:86]: Nice looks great!

**Comment by madsnedergaard** [comment]
Hey @VIKTORVAV99, I think this is really nicely done! 🙌 

I'm wondering though if we should move/concatenate some of these points here to the wiki? I personally find it easier to navigate (and update) the wiki - and we can go more in detail on specific topics and more easily separate things.

There's also some of the points here that are already covered in the wiki (like [local env setup](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Set-up-local-environment),
[parser requirements](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Technical-requirements-for-parser-data), [formatting](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Format-your-code-contribution)), although most could use a makeover and we should probably also restructure the pages!

That way we also keep this "intro" file brief and easy to get an overview of :)

**Comment by VIKTORVAV99** [comment]
> Hey @VIKTORVAV99, I think this is really nicely written! 🙌
> 
> I'm wondering though if we should move/concatenate some of these points here to the wiki? I personally find it easier to navigate (and update) the wiki - and we can go more in detail on specific topics and more easily separate things.
> 
> There's also some of the points here that are already covered in the wiki (like [local env setup](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Set-up-local-environment), [parser requirements](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Technical-requirements-for-parser-data), [formatting](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Format-your-code-contribution)), although most could use a makeover and we should probably also restructure the pages!
> 
> That way we also keep this "general contributing guidelines" more brief and easy to get an overview of :) Maybe we should (it can be later though) also add a small diagram or explanation of how the app works and why data is trustworthy 😄
> 
> And I'm sorry for only bringing it up now, didn't really think it through earlier 😬

I would love to bring more of these into the wiki but I'm not sure if it's something we should do right now as any changes would be instantaneous. We should probably wait until after the release is live and move over things from here section by section (and expanding on them).

So I think we should merge this right after the app merger and then we clean up the wiki to match the new structures and features of the app. Does that sound like a good plan? Julien also had some requests to crosslink more of our pages like the readme, this document and the wiki so we could do that at the same time as well.

File: CONTRIBUTING.md
Comment (madsnedergaard) [inline CONTRIBUTING.md:6]: Maybe we should also use this file to share a bit more about:
- When to create an issue vs discussion
- How we like PRs (small, fixing one thing, documented/explained in description, etc.)

I personally think https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md has some nice phrasings for inspiration :)

**Review by madsnedergaard — COMMENTED** [review]
Two minor comments in addition to the big one 😅

**Comment by madsnedergaard** [comment]
> > Hey @VIKTORVAV99, I think this is really nicely written! 🙌
> > I'm wondering though if we should move/concatenate some of these points here to the wiki? I personally find it easier to navigate (and update) the wiki - and we can go more in detail on specific topics and more easily separate things.
> > There's also some of the points here that are already covered in the wiki (like [local env setup](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Set-up-local-environment), [parser requirements](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Technical-requirements-for-parser-data), [formatting](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Format-your-code-contribution)), although most could use a makeover and we should probably also restructure the pages!
> > That way we also keep this "general contributing guidelines" more brief and easy to get an overview of :) Maybe we should (it can be later though) also add a small diagram or explanation of how the app works and why data is trustworthy 😄
> > And I'm sorry for only bringing it up now, didn't really think it through earlier 😬
> 
> I would love to bring more of these into the wiki but I'm not sure if it's something we should do right now as any changes would be instantaneous. We should probably wait until after the release is live and move over things from here section by section (and expanding on them).
> 
> So I think we should merge this right after the app merger and then we clean up the wiki to match the new structures and features of the app. Does that sound like a good plan? Julien also had some requests to crosslink more of our pages like the readme, this document and the wiki so we could do that at the same time as well.

That's a great point, we can definitely do that! I guess the other two comments I left can also wait until after the great merge :)

**Comment by VIKTORVAV99** [comment]
> > > Hey @VIKTORVAV99, I think this is really nicely written! 🙌
> > > I'm wondering though if we should move/concatenate some of these points here to the wiki? I personally find it easier to navigate (and update) the wiki - and we can go more in detail on specific topics and more easily separate things.
> > > There's also some of the points here that are already covered in the wiki (like [local env setup](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Set-up-local-environment), [parser requirements](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Technical-requirements-for-parser-data), [formatting](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Format-your-code-contribution)), although most could use a makeover and we should probably also restructure the pages!
> > > That way we also keep this "general contributing guidelines" more brief and easy to get an overview of :) Maybe we should (it can be later though) also add a small diagram or explanation of how the app works and why data is trustworthy 😄
> > > And I'm sorry for only bringing it up now, didn't really think it through earlier 😬
> > 
> > 
> > I would love to bring more of these into the wiki but I'm not sure if it's something we should do right now as any changes would be instantaneous. We should probably wait until after the release is live and move over things from here section by section (and expanding on them).
> > So I think we should merge this right after the app merger and then we clean up the wiki to match the new structures and features of the app. Does that sound like a good plan? Julien also had some requests to crosslink more of our pages like the readme, this document and the wiki so we could do that at the same time as well.
> 
> That's a great point, we can definitely do that! I guess the other two comments I left can also wait until after the great merge :)

I'll take a look at that and see if we can bring over some of the changes even before the merger (and the comments). But beyond that I think we should wait with any other major changes.

**Comment by madsnedergaard** [comment]
Let's pick this up again and get it merged - let me know if I should help with merge conflicts or anything else :)

**Comment by VIKTORVAV99** [comment]
> Let's pick this up again and get it merged - let me know if I should help with merge conflicts or anything else :)

Jesus, why are there 2000+ merge conflicts...? 😅
If you want to fix it feel free to do so otherwise it might just be simpler to create a new PR.

**Comment by VIKTORVAV99** [comment]
This PR is finally back in shape, I will do some final changes once #5068 is merged then we can merge this as well.

**Comment by VIKTORVAV99** [comment]
This should be ready for a final review now 🙂

File: CONTRIBUTING.md
Comment (madsnedergaard) [inline CONTRIBUTING.md:54]: Since we require a fixed Python version, maybe it's better to point to pyenv like we do in the wiki?
> Ensure you are using Python 3.8 (We recommend using [pyenv](https://github.com/pyenv/pyenv) to handle different Python versions.)

File: CONTRIBUTING.md
Comment (madsnedergaard) [reply CONTRIBUTING.md:63]: Given your recent work on the parsers/examples, I would actually suggest removing the example here and pointing to that folder - that way there's fewer places where things might get out of sync of something changes  :)

File: CONTRIBUTING.md
Comment (madsnedergaard) [inline CONTRIBUTING.md:85]: This can be replaced by a link to the examples folder now

**Review by madsnedergaard — APPROVED** [review]
A few minor things, otherwise it's great! 🙌

File: CONTRIBUTING.md
Comment (VIKTORVAV99) [reply CONTRIBUTING.md:85]: The examples folder only have one example + the example parser so far. Wouldn't we want to wait until we have at least a few more examples in there?

File: CONTRIBUTING.md
Comment (VIKTORVAV99) [reply CONTRIBUTING.md:54]: Makes sense, I'll add/change it.

File: CONTRIBUTING.md
Comment (VIKTORVAV99) [reply CONTRIBUTING.md:85]: Let's replace it and I'll see if I can add the other examples this week.

**Comment by VIKTORVAV99** [comment]
Merging this now. But I think we need to look over the wiki + contributing.md + parser readme + examples to make sure they are all aligned and contain the all necessary information.

**Pull request merged**
